kernel: add kernel 6.12

larvacea opened the pull request:
**Description of changes:**

A draft PR to add kernel 6.12 to the kernel kit. If you wish to build the new packages, you must add them to the file `kits/bottlerocket-kernel-kit/Cargo.toml`. If you wish to consume the kernel, the variant Cargo.toml in `bottlerocket` must include the image feature:
```
erofs-root-partition = true
```
This can be added to the file near other image features (such as `xfs-data-partition`).

**Testing done:**



**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

larvacea committed:
kernel: add kernel 6.12

bcressey commented on packages/kernel-6.12/1006-Select-prerequistes-for-gpu-drivers.patch:
```diff
@@ -0,0 +1,29 @@
+ 	select APERTURE_HELPERS
+ 	select DRM_GEM_SHMEM_HELPER
+ 	select DRM_KMS_HELPER
++	select DRM_DISPLAY_DP_HELPER
++	select DRM_DISPLAY_HELPER
++        select DRM_KMS_HELPER
++	select DRM_SUBALLOC_HELPER
++        select DRM_TTM
++	select DRM_TTM_HELPER
```
Why this is a patch as opposed to additional lines in the config file?

Please fix the spacing if the patch is required.

larvacea replied:
It's a patch because these aren't available as independent configs.

larvacea replied:
I will check the status of these Kconfigs, in case they are now menu-visible and can be toggled independently in the new kernel.

larvacea replied:
This is my best guess approximation of the patch we carry in 6.1 to enable the modules we need for GPU drivers in that kernel. I have tried to select the options directly from `config-bottlerocket` without success.

bcressey commented on Cargo.toml:
```diff
@@ -5,8 +5,10 @@ members = [
+    "packages/kernel-6.12",
     "packages/kmod-5.15-nvidia",
     "packages/kmod-6.1-nvidia",
+    "packages/kmod-6.12-nvidia",
```
Please remove the `kmod-6.12-nvidia` package from this PR and let's handle it in a follow up PR.

It's too much change to review repeatedly if the kernel-devel handling in the 6.12 spec file changes as I expect.

bcressey commented on packages/kernel-6.12/Cargo.toml:
```diff
@@ -0,0 +1,27 @@
+sha512 = "18a717b3cb0fa6c19ca6e0567691b3349b92390f8ac6a83555efb7cf026782f20a98292fad10b40f50b2972c39a1d25bddf6c60f49bd98afdc9698e3107498bd"
+force-upstream = true
+
+[[package.metadata.build-package.external-files]]
+# Use latest-neuron-srpm-url.sh to get this.
+url = "https://yum.repos.neuron.amazonaws.com/aws-neuronx-dkms-2.18.12.0.noarch.rpm"
+sha512 = "4ed92e661d0ba368eaf8f60e1a68c202062a26819231fcfd42a5ff05d20ad2f34b82b23359a88e80eea22ee5d0056ad769b6febd5d7e7b161da0e36434ba2579"
+force-upstream = true
```
This is outdated, please align it with kernel 6.1.

bcressey commented on packages/kernel-6.12/2001-Resolve-static-const-compiler-warnings.patch:
```diff
@@ -0,0 +1,151 @@
+From 5d1dae633d16144e6fe21da2e70b179a0c009da3 Mon Sep 17 00:00:00 2001
+From: Martin Harriman <larvacea@mac.com>
+Date: Wed, 5 Mar 2025 20:37:48 +0000
+Subject: [PATCH] Resolve static const compiler warnings
```
Why is this needed for 6.12 but not for the other kernels? We use the same GCC and it should be the same Neuron sources.

larvacea replied:
It's probably because I haven't updated the Neuron SRPM. I will fix that and see.

larvacea replied:
The latest srpm still exhibits this behavior:
```
  #11 225.7   149 | const static int root_arch_node_attrs_info_tbl_cnt = sizeof(root_arch_node_attrs_info_tbl) / sizeof(nsysfsmetric_attr_info_t);
  #11 225.7       | ^~~~~
  #11 225.8 /home/builder/rpmbuild/BUILD/bottlerocket-kernel-6.12-6.12.20-build/neuron/v1/neuron_dhal_v1.c:999:1: error: ‘static’ is not at beginning of declaration [-Werror=old-style-declaration]
```
I don't know why gcc would compile the neuron srpm sources differently here than in 6.1, but we do still need to patch the neuron sources here. From the C language specification 6.11 future language directions section:

> The placement of a storage-class specifier other than at the beginning of the declaration specifiers in
a declaration is an obsolescent feature.

I quote this because it was news to me. I am used to the word-salad world where these can come in any order.

bcressey replied:
This patch doesn't appear to be applied by the spec file, so it seems like it can be dropped.

bcressey replied:
Let's rename the file so it doesn't have a numbered prefix, something like:
```
neuron-resolve-static-const-compiler-warnings.patch
```

Otherwise it looks like a packaging error where the patch is in the directory but not listed in the spec as `Patch2001`.

bcressey commented on packages/kernel-6.12/bootconfig-aws.conf:
```diff
@@ -0,0 +1,2 @@
+kernel.initcall_blacklist = vmd_drv_init, megasas_init, mpt3sas_init, pqi_init
```
This line shouldn't be necessary for either aws or vmware snippets if we're not building the additional metal-only drivers.

bcressey commented on packages/kernel-6.12/config-bottlerocket:
```diff
@@ -0,0 +1,255 @@
+CONFIG_EXT4_FS=y
+CONFIG_EROFS_FS=y
+
+# With EROFS, we do not use SQUASHFS
+# CONFIG_SQUASHFS is not set
```
Move this down with the other disabled filesystems.

bcressey commented on packages/kernel-6.12/config-bottlerocket:
```diff
@@ -0,0 +1,255 @@
+CONFIG_BLK_DEV_NVME=y
+CONFIG_NVME_CORE=y
+
+# kernel-hardening-checker recommendations
+# CONFIG_ACPI_TABLE_UPGRADE is not set
+CONFIG_ARCH_MMAP_RND_BITS=33
+CONFIG_ARCH_MMAP_RND_COMPAT_BITS=16
+# CONFIG_BLK_DEV_WRITE_MOUNTED is not set
+# CONFIG_CACHESTAT_SYSCALL is not set
+CONFIG_DEBUG_NOTIFIERS=y
+CONFIG_DEBUG_SG=y
+# CONFIG_EFI_CUSTOM_SSDT_OVERLAYS is not set
+CONFIG_EFI_DISABLE_PCI_DMA=y
+# CONFIG_HIBERNATION is not set
+CONFIG_INIT_ON_ALLOC_DEFAULT_ON=y
+CONFIG_INTEL_IOMMU_DEFAULT_ON=y
+CONFIG_INTEL_IOMMU_SVM=y
+CONFIG_IOMMU_DEFAULT_DMA_STRICT=y
+CONFIG_KFENCE_SAMPLE_INTERVAL=100
+CONFIG_KFENCE=y
+# CONFIG_LEGACY_TIOCSTI is not set
+# CONFIG_LIVEPATCH is not set
+# CONFIG_MAGIC_SYSRQ is not set
+CONFIG_MAGIC_SYSRQ_DEFAULT_ENABLE=176
+CONFIG_RANDOM_KMALLOC_CACHES=y
+CONFIG_RANDOMIZE_BASE=y
+CONFIG_RANDOMIZE_KSTACK_OFFSET_DEFAULT=y
+CONFIG_RESET_ATTACK_MITIGATION=y
+CONFIG_X86_USER_SHADOW_STACK=y
```
I'd like to pull these out and have them come in a follow-up PR, so it's easier to do bisect builds before and after the changes to check for performance impacts.

bcressey commented on packages/kernel-6.12/config-bottlerocket:
```diff
@@ -0,0 +1,255 @@
+# Turn off Xen backend
+# CONFIG_XEN_BACKEND is not set
+
+# Turn off i915
+# CONFIG_DRM_I915 is not set
```
Is this still necessary?

larvacea replied:
Apparently it is. Without this, the CONFIG_DRM_I915 driver does build.

bcressey commented on packages/kernel-6.12/config-bottlerocket:
```diff
@@ -0,0 +1,255 @@
+
+# dm-verity and enabling it on the kernel command line
+CONFIG_BLK_DEV_DM=y
+CONFIG_DAX=y
```
Don't think we need this - `CONFIG_DAX=y` specifically.

larvacea replied:
Leaving this in for the moment, as it changes the build for several dm-verity pieces.

bcressey commented on packages/kernel-6.12/config-bottlerocket:
```diff
@@ -0,0 +1,255 @@
+# CONFIG_DRM_VIRTIO_GPU is not set
+# CONFIG_VIRTIO_DMA_SHARED_BUFFER is not set
+
+# NVDIMM
+# CONFIG_LIBNVDIMM is not set
```
Is this needed if `CONFIG_DAX` is not set?

bcressey commented on packages/kernel-6.12/latest-kernel-srpm-url.sh:
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+cmd='dnf install -q -y --releasever=latest yum-utils && yumdownloader -q --releasever=latest --source --urls kernel6.12'
+docker run --rm amazonlinux:2023 sh -c "${cmd}" \
```
```suggestion
docker run --rm public.ecr.aws/amazonlinux/amazonlinux:2023 sh -c "${cmd}" \
```

bcressey commented on packages/kernel-6.12/latest-neuron-srpm-url.sh:
```diff
@@ -0,0 +1,9 @@
+dnf install -q -y --releasever=latest yum-utils &&
+dnf download -q --repofrompath neuron,https://yum.repos.neuron.amazonaws.com --repo=neuron --urls aws-neuronx-dkms
+"
+docker run --rm amazonlinux:2023 bash -c "${cmd}" \
```
```suggestion
docker run --rm public.ecr.aws/amazonlinux/amazonlinux:2023 bash -c "${cmd}" \
```

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -26,11 +26,22 @@ Source3: https://github.com/intel/Intel-Linux-Processor-Microcode-Data-Files/arc
 
+%package ec2
+Summary: Microcode for EC2 processors
+URL: https://github.com/bottlerocket-os/bottlerocket/tree/develop/packages/microcode
```
This URL is no longer correct and since we're touching the spec we should take the opportunity to fix it.

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -26,11 +26,22 @@ Source3: https://github.com/intel/Intel-Linux-Processor-Microcode-Data-Files/arc
+%description ec2
+%{summary}.
+
+License: LicenseRef-scancode-amd-linux-firmware-export AND LicenseRef-scancode-intel-mcu-2018
```
This is not the right location for a `License` header`, it should go below `Summary`.

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -26,11 +26,22 @@ Source3: https://github.com/intel/Intel-Linux-Processor-Microcode-Data-Files/arc
+%package ec2
+Summary: Microcode for EC2 processors
+URL: https://github.com/bottlerocket-os/bottlerocket/tree/develop/packages/microcode
+Requires: %{_cross_os}microcode-amd
```
AMD microcode should get the same treatment, while we're making this change. Either:
1. separate ec2 and other subpackages
2. or a single subpackages that provides the ec2 and other capabilities

Then this would become:
```
Requires: %{_cross_os}microcode-amd-ec2
```

And then the dependencies will be more obviously correct to the reader, rather than making them wonder why AMD is different.

larvacea replied:
Let me check the AMD microcode list against the latest AL srpm again. My recollection is that there are no excess AMD microcode files (that is, the non-EC2 list would be empty).

larvacea replied:
I verified that the "microcode-amd-other" category would be empty. There are five (small) files on the AMD side.

bcressey replied:
I'd still like to see the change I asked for above. Option (2) would avoid an empty package while still letting this change to:
```
Requires: %{_cross_os}microcode-amd-ec2
```

larvacea replied:
Done

bcressey commented on packages/kernel-6.12/1006-Select-prerequistes-for-gpu-drivers.patch:
```diff
@@ -0,0 +1,29 @@
+From 574ee6cf29164ce587f525434611a76e10e4cc6e Mon Sep 17 00:00:00 2001
+From: Martin Harriman <mharrimn@amazon.com>
+Date: Sun, 30 Mar 2025 22:36:38 +0000
+Subject: [PATCH] Select prerequistes for gpu drivers
```
typo, `prerequisites`

bcressey replied:
(note that fixing this should also change the patch file name)

bcressey commented on packages/kernel-6.12/1006-Select-prerequistes-for-gpu-drivers.patch:
```diff
@@ -0,0 +1,29 @@
+ 	select DRM_GEM_SHMEM_HELPER
+ 	select DRM_KMS_HELPER
++	select DRM_DISPLAY_DP_HELPER
++	select DRM_DISPLAY_HELPER
++   select DRM_KMS_HELPER
++	select DRM_SUBALLOC_HELPER
++   select DRM_TTM
++	select DRM_TTM_HELPER
```
Can we fix the spacing for these or is it somehow correct like this?

larvacea replied:
I can fix the spacing. I can also hate the ASCII tab character. Again. More. As I have for the last 50-ish years.

bcressey commented on packages/kernel-6.12/bootconfig-vmware.conf:
```diff
@@ -0,0 +1 @@
+kernel.initcall_blacklist = vmd_drv_init, megasas_init, mpt3sas_init, pqi_init
```
This can just be an empty file for VMware.

larvacea commented:
Builds fail because of a SHA mismatch with the Neuron srpm. I have updated the draft with the latest Neuron srpm for the 6.12 package.

bcressey commented on packages/kernel-6.12/kernel-6.12.spec:
```diff
@@ -0,0 +1,1392 @@
+# No FIPS submission
+Conflicts: %{_cross_os}image-feature(fips)
+
+
```
```suggestion
# No squashfs support, rely on erofs for compression
Conflicts: %{_cross_os}image-feature(no-erofs-root-partition)
```

bcressey commented on packages/kernel-6.12/kernel-6.12.spec:
```diff
@@ -0,0 +1,1392 @@
+%global debug_package %{nil}
```
If you don't disable strip like this, then module signatures end up stripped and modules can't be loaded at runtime.

```suggestion
%global debug_package %{nil}
%global __strip /bin/true
```

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -17,27 +17,40 @@ License: LicenseRef-scancode-amd-linux-firmware-export AND LicenseRef-scancode-i
+%description ec2
+%{summary}.
+
+# There is no amd-other subpackage because amd-ec2 includes all AMD microcode.
+%package amd-ec2
```
```suggestion
%package amd
Provides: %{_cross_os}microcode-amd-ec2
Provides: %{_cross_os}microcode-amd-other
```

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -17,27 +17,40 @@ License: LicenseRef-scancode-amd-linux-firmware-export AND LicenseRef-scancode-i
 
-%package amd
-Summary: Microcode for AMD processors
+# Microcode applicable to Intel and AMD processors in EC2
```
unnecessary comment if the package summary provides this info:
```suggestion
```

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -17,27 +17,40 @@ License: LicenseRef-scancode-amd-linux-firmware-export AND LicenseRef-scancode-i
-Summary: Microcode for AMD processors
+# Microcode applicable to Intel and AMD processors in EC2
+%package ec2
+Summary: Microcode for EC2 AMD and Intel processors
```
Rephrase so it doesn't sound like these are "EC2[-specific] AMD processors"
```suggestion
Summary: Microcode for AMD and Intel processors available in EC2
```

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -17,27 +17,40 @@ License: LicenseRef-scancode-amd-linux-firmware-export AND LicenseRef-scancode-i
+
+# There is no amd-other subpackage because amd-ec2 includes all AMD microcode.
+%package amd-ec2
+Summary: Microcode for EC2 AMD processors
```
```suggestion
Summary: Microcode for AMD processors
```

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -48,13 +61,23 @@ URL: https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git
-Summary: Microcode for Intel processors
+# Intel microcode for Intel processors not present in EC2.
+%package intel-other
+Summary: Microcode for Intel processors (not EC2)
```
```suggestion
Summary: Microcode for Intel processors not available in EC2
```

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -48,13 +61,23 @@ URL: https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git
 
-%package intel
-Summary: Microcode for Intel processors
+# Intel microcode for Intel processors not present in EC2.
```
```suggestion
```

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -48,13 +61,23 @@ URL: https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git
+%{summary}.
+
+%package intel-ec2
+Summary: Microcode for Intel processors on EC2
```
```suggestion
Summary: Microcode for Intel processors available in EC2
```

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -116,18 +139,173 @@ install -p -m 0644 intel/intel-ucode/* %{buildroot}%{_cross_libdir}/firmware/int
 %files
 
-%files amd
+%files amd-ec2
```
```suggestion
%files amd
```

bcressey commented on packages/microcode/microcode.spec:
```diff
@@ -17,27 +17,40 @@ License: LicenseRef-scancode-amd-linux-firmware-export AND LicenseRef-scancode-i
 Requires: %{_cross_os}microcode-amd-license
 
-%description amd
+%description amd-ec2
```
```suggestion
%description amd
```

arnaldo2792 commented:
> Builds fail because of a SHA mismatch with the Neuron srpm. I have updated the draft with the latest Neuron srpm for the 6.12 package.

Could you please pull this change out of your PR? As this will help with others that are facing the same problem.

bcressey commented on packages/kernel-6.12/config-bottlerocket:
```diff
@@ -0,0 +1,224 @@
+CONFIG_EXT4_FS=y
+CONFIG_EROFS_FS=y
+
+
```
remove extra newline
```suggestion
```

bcressey commented on packages/kernel-6.12/config-bottlerocket:
```diff
@@ -0,0 +1,226 @@
+CONFIG_EROFS_FS=y
+
+
+# ARM: EFI_ZBOOT
```
```suggestion
# Compress arm64 kernels
```

bcressey commented on packages/kernel-6.12/config-bottlerocket:
```diff
@@ -0,0 +1,226 @@
+# CONFIG_DRM_BOCHS is not set
+# CONFIG_SYSFB_SIMPLEFB is not set
+
+# With 6.1 some of the functionalities used by the nvidia driver have moved behind
+# some extra config options CONFIG_DRM_KMS_HELPER and CONFIG_DRM_DISPLAY_HELPER.
+# These config options can not be selected individually, but are selected by certain
+# drivers. Enable the SIMPLEDRM driver, which is a minimal drm driver enabling
+# those helpers for platform provided framebuffers.
+CONFIG_DRM_SIMPLEDRM=y
+CONFIG_DRM_DISPLAY_HELPER=y
+CONFIG_DRM_TTM_HELPER=y
```
This can be cleaned up and simplified because of the patch:
```suggestion
# Enable the SIMPLEDRM driver, which is a minimal drm driver that enables the helpers for
# platform provided framebuffers required by the NVIDIA driver.
CONFIG_DRM_SIMPLEDRM=y
```

bcressey commented on packages/kernel-6.12/config-bottlerocket:
```diff
@@ -0,0 +1,226 @@
+# CONFIG_DRM_VIRTIO_GPU is not set
+# CONFIG_VIRTIO_DMA_SHARED_BUFFER is not set
+
+# CONFIG_LIBNVDIMM is not set
```
Add a comment explaining why

bcressey commented on packages/kernel-6.12/kernel-6.12.spec:
```diff
@@ -0,0 +1,1395 @@
+%if "%{_cross_arch}" == "x86_64"
+Requires: (%{name}-modules-neuron if (%{_cross_os}variant-platform(aws) without %{_cross_os}variant-flavor(nvidia)))
+%endif
+
```
add this back since eventually FIPS will be supported:
```suggestion

# Pull in FIPS-related files if needed.
Requires: (%{name}-fips if %{_cross_os}image-feature(fips))

```

bcressey commented on packages/kernel-6.12/kernel-6.12.spec:
```diff
@@ -0,0 +1,1395 @@
+rpmkeys --checksig %{S:2} --dbpath "${PWD}/rpmdb"
+rm -rf "${PWD}/rpmdb"
+rpm2cpio %{S:2} | cpio -idmu './usr/src/aws-neuronx-*'
+
```
```suggestion
```

bcressey commented on packages/kernel-6.12/kernel-6.12.spec:
```diff
@@ -0,0 +1,1395 @@
+%{_cross_kmoddir}/modules.symbols.bin
+%{_cross_kmoddir}/modules.weakdep
+%{_cross_kmoddir}/System.map
+%if "%{_cross_arch}" == "x86_64"
```
restore this newline:
```suggestion

%if "%{_cross_arch}" == "x86_64"
```

bcressey commented on packages/kernel-6.12/kernel-6.12.spec:
```diff
@@ -0,0 +1,1395 @@
+%{_cross_kmoddir}/kernel/arch/x86/kvm/kvm.%{_ko}
+%{_cross_kmoddir}/kernel/arch/x86/platform/intel/iosf_mbi.%{_ko}
+%endif
+
```
remove this newline:
```suggestion
```

bcressey reviewed (commented):
Some minor notes to clean up the diff between 6.1 and 6.12 specs:
```
git diff --no-index packages/kernel-6.1/kernel-6.1.spec packages/kernel-6.12/kernel-6.12.spec
```

larvacea committed:
microcode: create subpackages

Create subpackages to allow kernels to consume just the files relevant
for processors available in EC2.

bcressey commented on packages/kernel-6.12/config-bottlerocket:
```diff
@@ -0,0 +1,221 @@
+# CONFIG_DRM_BOCHS is not set
+# CONFIG_SYSFB_SIMPLEFB is not set
+
+# Enable the SIMPLEDRM driver, which is a minimal drm driver enabling
+# those helpers for platform provided framebuffers.
+CONFIG_DRM_SIMPLEDRM=m
```
This doesn't provide the crucial bit of context from my suggestion, which is that we're doing this for the NVIDIA driver:
```suggestion
# Enable the SIMPLEDRM driver, which is a minimal drm driver that enables the helpers for
# platform provided framebuffers required by the NVIDIA driver.
CONFIG_DRM_SIMPLEDRM=m
```

larvacea replied:
Done (but word-wrapped to 80 characters).

arnaldo2792 commented on packages/kernel-6.12/kernel-6.12.spec:
```diff
@@ -0,0 +1,1397 @@
+  (( i+=1 ))
+done
+
+LOWERPATH=$(systemd-escape --path %{_cross_sharedstatedir}/kernel-devel/.overlay/lower)
+mkdir -p %{buildroot}%{_cross_unitdir}/"${LOWERPATH}.mount.d"
+sed -e 's|PREFIX|%{_cross_prefix}|g' %{S:210} \
+  > %{buildroot}%{_cross_unitdir}/"${LOWERPATH}.mount.d"/no-squashfs.conf
```
It took me a bit to piece these together. We should consider moving the kernel-devel mount files from the core kit to the kernel kit. It feels a bit disconnected that we are dropping a config here and expect it to be picked up. If we provide the unit files to configure the kernel sources through this package, it will be less disconnected. Do you have opinions on this @bcressey ?

arnaldo2792 commented on packages/kernel-6.12/kernel-6.12.spec:
```diff
@@ -0,0 +1,1397 @@
+%global _cross_kimage vmlinuz.efi
+%endif
+
+%global _ko ko
```
nit: a comment that here that describes why the macro would be nice

arnaldo2792 commented on Cargo.toml:
```diff
@@ -5,6 +5,7 @@ members = [
     "packages/grub",
     "packages/kernel-5.15",
     "packages/kernel-6.1",
+    "packages/kernel-6.12",
```
I'm posting this message here for the lack of a better place to leave this commit, commits are missing the `signed-off` footer.

rpkelly commented on packages/kernel-6.12/Cargo.toml:
```diff
@@ -0,0 +1,27 @@
+# Use latest-neuron-srpm-url.sh to get this.
+url = "https://yum.repos.neuron.amazonaws.com/aws-neuronx-dkms-2.20.28.0.noarch.rpm"
+sha512 = "e94ea8244f41fd546fb2aa66a793488069a26724b59a4d05355011fca2643f0874a2971e617563c5c2c402952ed160c3b85548f6013c2c60fbc41c89b39133d8"
+force-upstream = true
```
This should be changed to our mirror.

rpkelly commented on packages/kernel-6.12/kernel-6.12.spec:
```diff
@@ -0,0 +1,1397 @@
+Source0: hhttps://cdn.amazonlinux.com/al2023/blobstore/e48aacb6891d5dc0f536065fa7b52d7b8e7bc0d809fe6f93043b7e8bb0e1fe3b/kernel6.12-6.12.20-23.97.amzn2023.src.rpm
+Source1: gpgkey-B21C50FA44A99720EAA72F7FE951904AD832C631.asc
+# Use latest-neuron-srpm-url.sh to get this.
+Source2: https://yum.repos.neuron.amazonaws.com/aws-neuronx-dkms-2.20.28.0.noarch.rpm
```
See above about using a mirror

larvacea commented:
Updated Cargo.toml to fetch the Neuron source rpm from the Bottlerocket lookaside cache.

ginglis13 commented on packages/kernel-6.12/neuron-resolve-static-const-compiler-warnings.patch:
```diff
@@ -0,0 +1,151 @@
+From 5d1dae633d16144e6fe21da2e70b179a0c009da3 Mon Sep 17 00:00:00 2001
```
Is there a reason we haven't named this patch prefixed with `1008-` ?

larvacea replied:
Yes: this patch isn't applied when we autopatch the world with the 1000-series patches. It gets applied explicitly as part of unpacking and building the neuron driver.

ginglis13 reviewed (approved):
Nice!
